diff -bdur '--exclude-from=../make-diffs-ignore-public.txt' --new-file server/finish/src/app.js client/jquery/src/app.js
--- server/finish/src/app.js	2017-01-06 21:06:19.211000000 -0500
+++ client/jquery/src/app.js	2017-01-04 16:06:15.658000000 -0500
@@ -39,4 +39,6 @@
   .then(() => console.log('messages table cleared.'))
   .catch(err => console.log('ERROR clearing messages table:', err));
 
+//app.service('/messages').Model.remove({}, { multi: true })
+
 module.exports = app;
diff -bdur '--exclude-from=../make-diffs-ignore-public.txt' --new-file server/finish/src/services/message/hooks/index.js client/jquery/src/services/message/hooks/index.js
--- server/finish/src/services/message/hooks/index.js	2017-01-15 15:14:05.848000000 -0500
+++ client/jquery/src/services/message/hooks/index.js	2017-01-15 16:44:12.847000000 -0500
@@ -6,7 +6,7 @@
 
 const restrictToSender = require('./restrict-to-sender');
 const process = require('./process');
-const { setCreatedAt, populate, dePopulate, serialize } = require('feathers-hooks-common');
+const { setCreatedAt, populate, dePopulate, serialize, when, isProvider } = require('feathers-hooks-common');
 
 const restrictToSenderOrServer = when(isProvider('external'), restrictToSender());
 
