--- 01/hooks/1.js	2017-01-13 14:03:07.126000000 -0500
+++ 01/hooks/2.js	2017-01-13 14:06:55.419000000 -0500
@@ -1,5 +1,5 @@
 
-// Example - Use hooks with service
+// Example - Use soft deletes with a service
 
 const expressServerConfig = require('../common/expressServerConfig');
 const expressMiddleware = require('../common/expressMiddleware');
@@ -35,10 +35,12 @@
   const userService = app.service('users');
   
   const {
+    softDelete, when,
     setCreatedAt, setUpdatedAt, unless, remove /* , validateSchema */
   } = commonHooks;
   
   userService.before({
+    all: when(hook => hook.method !== 'find', softDelete()),
     create: [
       validateSchema(userSchema(), Ajv),
       authHooks.hashPassword(),
