--- 01/rest/2.js	2017-01-13 12:42:56.178000000 -0500
+++ 01/websocket/1.js	2017-01-13 13:31:34.853000000 -0500
@@ -1,15 +1,17 @@
 
-// Example - Create REST API, and serve static files
+// Example - Create REST & socketio API, and serve static files
 
 const expressServerConfig = require('../common/expressServerConfig');
 const expressMiddleware = require('../common/expressMiddleware');
 const rest = require('feathers-rest');
+const socketio = require('feathers-socketio');
 const NeDB = require('nedb');
 const path = require('path');
 const service = require('feathers-nedb');
 
 const app = expressServerConfig()
   .configure(rest())
+  .configure(socketio())
   .configure(services)
   .configure(expressMiddleware);
 
