diff -bdur --new-file 02/gen2/src/models/teams.model.js 02/gen3/src/models/teams.model.js
--- 02/gen2/src/models/teams.model.js	1969-12-31 19:00:00.000000000 -0500
+++ 02/gen3/src/models/teams.model.js	2017-04-06 12:47:01.036496074 -0400
@@ -0,0 +1,12 @@
+const NeDB = require('nedb');
+const path = require('path');
+
+module.exports = function(app) {
+  const dbPath = app.get('nedb');
+  const Model = new NeDB({
+    filename: path.join(dbPath, 'teams.db'),
+    autoload: true
+  });
+
+  return Model;
+};
diff -bdur --new-file 02/gen2/src/services/index.js 02/gen3/src/services/index.js
--- 02/gen2/src/services/index.js	2017-04-06 12:34:15.844496074 -0400
+++ 02/gen3/src/services/index.js	2017-04-06 12:47:01.036496074 -0400
@@ -2,7 +2,10 @@
 
 const users = require('./users/users.service.js');
 
+const teams = require('./teams/teams.service.js');
+
 module.exports = function() {
   const app = this; // eslint-disable-line no-unused-vars
   app.configure(users);
+  app.configure(teams);
 };
diff -bdur --new-file 02/gen2/src/services/teams/teams.filters.js 02/gen3/src/services/teams/teams.filters.js
--- 02/gen2/src/services/teams/teams.filters.js	1969-12-31 19:00:00.000000000 -0500
+++ 02/gen3/src/services/teams/teams.filters.js	2017-04-06 12:47:01.040496074 -0400
@@ -0,0 +1,6 @@
+/*eslint no-console: 1 */
+console.warn('You are using the default filter for the teams service. For more information about event filters see https://docs.feathersjs.com/api/events.html#event-filtering'); // eslint-disable-line no-console
+
+module.exports = function(data, connection, hook) { // eslint-disable-line no-unused-vars
+  return data;
+};
diff -bdur --new-file 02/gen2/src/services/teams/teams.hooks.js 02/gen3/src/services/teams/teams.hooks.js
--- 02/gen2/src/services/teams/teams.hooks.js	1969-12-31 19:00:00.000000000 -0500
+++ 02/gen3/src/services/teams/teams.hooks.js	2017-04-06 12:47:01.036496074 -0400
@@ -0,0 +1,35 @@
+'use strict';
+
+const { authenticate } = require('feathers-authentication').hooks;
+
+module.exports = {
+  before: {
+    all: [ authenticate('jwt') ],
+    find: [],
+    get: [],
+    create: [],
+    update: [],
+    patch: [],
+    remove: []
+  },
+
+  after: {
+    all: [],
+    find: [],
+    get: [],
+    create: [],
+    update: [],
+    patch: [],
+    remove: []
+  },
+
+  error: {
+    all: [],
+    find: [],
+    get: [],
+    create: [],
+    update: [],
+    patch: [],
+    remove: []
+  }
+};
diff -bdur --new-file 02/gen2/src/services/teams/teams.service.js 02/gen3/src/services/teams/teams.service.js
--- 02/gen2/src/services/teams/teams.service.js	1969-12-31 19:00:00.000000000 -0500
+++ 02/gen3/src/services/teams/teams.service.js	2017-04-06 12:47:01.032496074 -0400
@@ -0,0 +1,31 @@
+'use strict';
+
+// Initializes the `teams` service on path `/teams`
+const createService = require('feathers-nedb');
+const createModel = require('../../models/teams.model');
+const hooks = require('./teams.hooks');
+const filters = require('./teams.filters');
+
+module.exports = function() {
+  const app = this;
+  const Model = createModel(app);
+  const paginate = app.get('paginate');
+
+  const options = {
+    name: 'teams',
+    Model,
+    paginate
+  };
+
+  // Initialize our service with any options it requires
+  app.use('/teams', createService(options));
+
+  // Get our initialized service so that we can register hooks and filters
+  const service = app.service('teams');
+
+  service.hooks(hooks);
+
+  if (service.filter) {
+    service.filter(filters);
+  }
+};
diff -bdur --new-file 02/gen2/test/services/teams.test.js 02/gen3/test/services/teams.test.js
--- 02/gen2/test/services/teams.test.js	1969-12-31 19:00:00.000000000 -0500
+++ 02/gen3/test/services/teams.test.js	2017-04-06 12:47:01.044496074 -0400
@@ -0,0 +1,12 @@
+'use strict';
+
+const assert = require('assert');
+const app = require('../../src/app');
+
+describe('\'teams\' service', () => {
+  it('registered the service', () => {
+    const service = app.service('teams');
+
+    assert.ok(service, 'Registered the service');
+  });
+});
