--- 02/gen3/src/services/teams/teams.hooks.js	2017-04-06 12:47:01.036496074 -0400
+++ 02/gen4/src/services/teams/teams.hooks.js	2017-04-06 12:48:06.780496074 -0400
@@ -2,6 +2,8 @@
 
 const { authenticate } = require('feathers-authentication').hooks;
 
+const populateTeams = require('../../hooks/populate-teams');
+
 module.exports = {
   before: {
     all: [ authenticate('jwt') ],
@@ -15,7 +17,7 @@
 
   after: {
     all: [],
-    find: [],
+    find: [populateTeams()],
     get: [],
     create: [],
     update: [],
