--- 02/app1/src/services/teams/hooks/index.js	2017-03-29 08:57:36.706003415 -0400
+++ 02/service/src/services/teams/hooks/index.js	2017-03-29 08:57:36.710005414 -0400
@@ -3,6 +3,23 @@
 const globalHooks = require('../../../hooks');
 const hooks = require('feathers-hooks');
 const auth = require('feathers-authentication').hooks;
+const { populate, serialize } = require('feathers-hooks-common'); // added
+
+const populateSchema = { // added
+  include: [{
+    service: 'users',
+    nameAs: 'members',
+    parentField: 'memberIds',
+    childField: '_id'
+  }]
+};
+
+const serializeSchema = { // added
+  exclude: ['_id', '_include'],
+  members: {
+    only: ['email', 'role']
+  }
+};
 
 exports.before = {
   all: [
@@ -20,8 +37,8 @@
 
 exports.after = {
   all: [],
-  find: [],
-  get: [],
+  find: [ populate({ schema: populateSchema }), serialize(serializeSchema) ], // changed
+  get: [ populate({ schema: populateSchema }), serialize(serializeSchema) ], // changed
   create: [],
   update: [],
   patch: [],
